File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        HashSet<Integer> setA=new HashSet<>();
        HashSet<Integer> setB=new HashSet<>();
        int[] result=new int[A.length];
        for(int i=0;i<A.length;i++){
            setA.add(A[i]);
            setB.add(B[i]);
            HashSet intersection=new HashSet<>(setA);
            intersection.retainAll(setB);
            result[i]=intersection.size();
        }
        return result;
    }
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        hash=dict()
        result=[]
        for i in range(len(A)):
            #print(i)
            hash[A[i]]=hash.get(A[i],0)+1
            hash[B[i]]=hash.get(B[i],0)+1
            count=0
            if hash[A[i]]==2:
                count+=1
            if hash[B[i]]==2 and A[i]!=B[i]:
                count+=1
            if i==0:
                result.append(count)
            else:
                #print('chckpt1',result[-1],result)
                result.append(count+result[-1])
        return result
